modules/update-best-practices.adoc

```diff
@@ -0,0 +1,57 @@
+// Module included in the following assemblies:
+//
+// * updating/preparing_for_updates/updating-cluster-prepare.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="update-best-practices_{context}"]
+= Best practices for cluster updates
+
+{product-title} is designed to provide a robust update experience that allows clusters to update with minimal disruptions to workloads.
+Updates will not begin unless the cluster is determined to be in an upgradeable state at the time of the update request.
+
+While this design helps ensure that updates are successful as long as some key conditions are met, there are a number of actions you can take to increase your chances of a successful cluster update.
+
+[discrete]
+[id="recommended-versions_{context}"]
+=== Choose versions recommended by the OpenShift Update Service
+
+The OpenShift Update Service (OSUS) provides update recommendations based on cluster characteristics such as the cluster's subscribed channel, which are then saved by the Cluster Version Operator as either recommended or conditional updates.
+While it is possible to attempt an update to a version that is not recommended by OSUS, doing so significantly increases the risk of update failure or unintended consequences to the cluster after the update has finished.
+
+Choose only update targets that are recommended by OSUS to ensure a successful update.
+
+[discrete]
+[id="critical-alerts_{context}"]
+=== Address all critical alerts on the cluster
+
+Critical alerts must always be addressed as soon as possible, but it is especially important to address these alerts and resolve any problems before initiating a cluster update.
+Failing to address critical alerts before beginning an update can cause a loss of data or a major failure of cluster services.
+
+You should also periodically review Warning and Info alerts on the cluster to address any potentially problematic conditions before initiating an update.
+
+[discrete]
+[id="cluster-upgradeable_{context}"]
+=== Ensure that the cluster is in an Upgradable state
+
+When one or more Operators have not reported their `Upgradeable` condition as `true` for more than an hour, the `ClusterNotUpgradeable` warning alert is triggered in the cluster.
+In most cases patch updates are not blocked by this alert, but you cannot perform a minor version update until this alert is resolved and all Operators report `Upgradeable` as `true`.
+
+[discrete]
+[id="nodes-ready_{context}"]
+=== Ensure that all nodes are available
```

Comment:
I think "all nodes are available" is not exactly the condition that we want. The point is that the cluster should not be running with little to no spare capacity. If the cluster is "full", then taking down a node to upgrade it can be either slow  or even impossible. So I would describe this section as "make sure the cluster has enough spare capacity so that the upgrade process can actually take some nodes offline to update them. This has two flavors, (you already mention one of them):
1. There is spare capacity on compute nodes, so that when taking a node offline temporarily, its workloads can be placed on other nodes comfortably
2. Worker pools do not have enough unavailable notes to prevent taking more nodes offline temporarily.

Reply:
We have noticed that many times when a node is down MCO gets stuck. So I am supportive of the idea. I also agree that we should have spare nodes which would give some breathing room to pods.